lib/travis/guest-api/reporter.rb

```diff
@@ -61,9 +64,28 @@ def close
       # simple helpers
       def send_log(job_id, output, last_message = false)
         @logs_part_number += 1
-        message = { id: job_id, log: output, number: @logs_part_number }
-        message[:final] = true if last_message
-        notify('job:test:log', message)
+
+        @jobs_messages_sizes[job_id] ||= {}
+        @jobs_messages_sizes[job_id][:size] ||= 0
+        @jobs_messages_sizes[job_id][:size] += output.size
+
+        max_log_size = (Travis.config.max_log_size || 1024*5)
```

Comment:
the "OR branch" is pointless because Travis::Config will throw an exception if the config is missing from config file